src/core/server/graph/plugins/helpers.ts

```diff
@@ -108,7 +108,13 @@ function hoistCoralErrorExtensions(
 
   // Hoist the message from the original error into the message of the base
   // error.
-  (err as any).message = extensions.message;
+  const rawError = err as any;
+  rawError.message = extensions.message;
+  rawError.traceID = ctx.id;
+
+  if (err.extensions) {
+    err.extensions.traceID = ctx.id;
+  }
```

Comment:
We don't need to attach the traceID to the error directly. Extensions are always available so we can rely on that. This will be more spec compilant.

Should put implementation in `serializeExtensions` e.g. in line 107 it should say 
```ts
const extensions = originalError.serializeExtensions(bundle, traceID);
```

Reply:
That actually makes a lot of sense!

Reply:
This does not work. Also modifying the `serializeExtensions` method to receive the traceID and set it in the returned object, but that also does not work. Am fairly certain I need to set this here.

Reply:
Hmm I just tried it and it worked, let's chat about this.